v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/schema/typemapping/provider/MysqlMappingProviderTest.java

```diff
@@ -0,0 +1,106 @@
+public class MysqlMappingProviderTest {
+  @Test
+  public void testMySqlMappingProvider() {
+    Long[] testMods = {1L, 1L};
+    // We don't have a use case for arrays yet.
+    Long[] testArrayBounds = {};
+    ImmutableMap<String, String> stringifiedMapping =
+        MysqlMappingProvider.getMapping().entrySet().stream()
+            .map(
+                e ->
+                    Map.entry(
+                        e.getKey(),
+                        e.getValue()
+                            .getSchema(testMods, testArrayBounds)
+                            .toString()
+                            .replaceAll("\\s+", "")))
+            .collect(ImmutableMap.toImmutableMap(Entry::getKey, Entry::getValue));
+    assertThat(stringifiedMapping).isEqualTo(expectedMapping());
+  }
+
+  private ImmutableMap<String, String> expectedMapping() {
+    return ImmutableMap.<String, String>builder()
+        .put("BIGINT", "\"long\"")
+        .put(
+            "BIGINT UNSIGNED",
+            "{\"type\":\"bytes\",\"logicalType\":\"decimal\",\"precision\":1,\"scale\":1}")
+        .put("BINARY", "\"string\"")
+        .put("BIT", "\"long\"")
+        .put("BLOB", "\"string\"")
+        .put("BOOL", "\"int\"")
+        .put("CHAR", "\"string\"")
+        .put("DATE", "{\"type\":\"long\",\"logicalType\":\"timestamp-micros\"}")
+        .put(
+            "DateTime",
+            "{\"type\":\"record\","
+                + "\"name\":\"datetime\","
+                + "\"fields\": ["
+                + "{\"name\":\"date\",\"type\":{\"type\":\"int\",\"logicalType\":\"date\"}},"
+                + "{\"name\":\"time\",\"type\":{\"type\":\"long\",\"logicalType\":\"time-micros\"}}]}")
+        .put(
+            "DECIMAL",
+            "{\"type\":\"bytes\",\"logicalType\":\"decimal\",\"precision\":1,\"scale\":1}")
+        .put("DOUBLE", "\"double\"")
+        .put("ENUM", "\"string\"")
+        .put("FLOAT", "\"float\"")
+        .put("INTEGER", "\"int\"")
+        .put("Json", "{\"type\":\"string\",\"logicalType\":\"json\"}")
+        .put("LONGBLOB", "\"string\"")
+        .put("LONGTEXT", "\"string\"")
+        .put("MEDIUMBLOB", "\"string\"")
+        .put("MEDIUMINT", "\"int\"")
+        .put("MEDIUMTEXT", "\"string\"")
+        .put("SET", "\"string\"")
+        .put("SMALLINT", "\"int\"")
+        .put("TEXT", "\"string\"")
+        .put(
+            "TIME",
+            "{\"type\":\"record\","
+                + "\"name\":\"interval\","
+                + "\"fields\":["
+                + "{\"name\":\"months\",\"type\":\"int\"},"
+                + "{\"name\":\"hours\",\"type\":\"int\"},"
+                + "{\"name\":\"micros\",\"type\":\"long\"}]}")
+        .put("TIMESTAMP", "{\"type\":\"long\",\"logicalType\":\"timestamp-micros\"}")
+        .put("TINYBLOB", "\"string\"")
+        .put("TINYINT", "\"int\"")
+        .put("TINYTEXT", "\"string\"")
+        .put("VARBINARY", "\"string\"")
+        .put("VARCHAR", "\"string\"")
```

Comment:
Should'nt varchar also be displays the mods like the other types?

Reply:
For MySQL varchar is mapped to string of infinite length -  https://cloud.google.com/datastream/docs/unified-types#:~:text=STRING%20(hex%20encoded)-,VARCHAR,STRING,-YEAR